### ISSUE TITLE
[API] Not all members (of our org) are returned from our `/users` endpoint

### ISSUE THREAD

**Issue by iautom8things** [getsentry/sentry#66147]
### Environment

SaaS (https://sentry.io/)

### Steps to Reproduce

Call GET `https://sentry.io/api/0/organizations/chapterspot/users/` this returns 22 users.
Call GET `https://sentry.io/api/0/organizations/chapterspot/members/` this returns 25 members.

The 3 users/members are shown on the UI.

### Expected Result

These (I think?) should return the same members.  I honestly don't understand what the difference is between a user and a member from this context.  These users are members of our org.

### Actual Result

I was directed to open this issue from the Discord server.  I don't feel comfortable sharing details about our users.  I'd be happy to provide more details in a less open venue.

### Product Area

APIs

### Link

https://sentry.io/api/0/organizations/chapterspot/users/

**Comment by Dhrumil-Sentry** [comment]
Thanks for this report, we're investigating this

**Comment by sentaur-athena** [comment]
@iautom8things the `/members/` endpoint is what you want to use here. I'll update our API docs to make it clear that the `/users/` endpoint is mostly used for project based filtering.

**Comment by iautom8things** [comment]
> @iautom8things the `/members/` endpoint is what you want to use here. I'll update our API docs to make it clear that the `/users/` endpoint is mostly used for project based filtering.

Thanks. I realized that we needed to use the /members endpoint. But even with your message it's still not clear to me what the difference is. What do you mean it's for project based filtering? For our case it's not clear what's special about the three people not returned by the /users endpoint. What's different about their membership to our org?

**Comment by sentaur-athena** [comment]
I'm still investigating for your org specifically but here are some overall differences between the 2:
1. `members` also returns the invited members who have not joined yet while `users` doesn't
2. `users` should not be used really for this purpose and I'll be updating our API docs to reflect it. If you want to fetch users under a specific project it would be useful but for example members who don't belong to a project won't show up.

**Comment by iautom8things** [comment]
> I'm still investigating for your org specifically but here are some overall differences between the 2:
> 
> 1. `members` also returns the invited members who have not joined yet while `users` doesn't
> 
> 2. `users` should not be used really for this purpose and I'll be updating our API docs to reflect it. If you want to fetch users under a specific project it would be useful but for example members who don't belong to a project won't show up. 

Gotchya, thank you for the explanation! I think that second point is likely what was happening!

**Comment by sentaur-athena** [comment]
Yeah, I verified that for one of the missing members. Please continue using `/members/` endpoint. Thank you.
